keps/sig-scheduling/5471-enable-sla-based-scheduling/README.md

```diff
@@ -0,0 +1,1430 @@
+
+**Example Configuration:**
+
+```yaml
+# Spot nodes with 80% SLA get a repelling taint
+apiVersion: v1
+kind: Node
+metadata:
+  name: spot-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/sla
+    value: "800"
+    effect: NoSchedule
+---
+# Cost-optimized workload explicitly tolerates SLA >= 750
+apiVersion: v1
+kind: Pod
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Gt
+    value: "750"
+    effect: NoSchedule
+```
+
+#### Story 2 — AI inference service with strict SLOs
+
+As an AI platform engineer, I want to ensure my latency-critical inference pods only run on nodes with SLA ≥ 95%, and I want them to be evicted if the node's SLA rating drops below that threshold.
+
+Taints and tolerations with numeric comparisons give me this eviction capability, which NodeAffinity cannot provide.
+
+**Example Configuration:**
+
+```yaml
+# High-SLA on-demand node
+apiVersion: v1
+kind: Node
+metadata:
+  name: ondemand-node-1
+spec:
+  taints:
+  - key: node.kubernetes.io/sla
+    value: "950"
+    effect: NoExecute
+---
+# Inference service requires SLA >= 950 with 30s grace period
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inference-service
+spec:
+  template:
+    spec:
+      tolerations:
+      - key: node.kubernetes.io/sla
+        operator: Gt
+        value: "950"
+        effect: NoExecute
+        tolerationSeconds: 30
+```
+
+#### Story 3 — AI training workload balancing cost and reliability
+
+As an ML engineer running large distributed training, I want to run most worker pods on cheaper spot GPU nodes, but keep certain roles (e.g., parameter servers, checkpoint writers) on SLA ≥ 99.9% on-demand GPUs.
+
+With numeric tolerations, I can opt-in only the pods that are safe to run on spot, while letting the cluster's default taints repel all others.
+
+**Example Configuration:**
+
+```yaml
+# Parameter server requires ultra-high reliability
+apiVersion: v1
+kind: Pod
+metadata:
+  name: parameter-server
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Gt
+    value: "999"  # 99.9% SLA
+    effect: NoSchedule
+  containers:
+  - name: ps
+    resources:
+      requests:
+        nvidia.com/gpu: 1
+---
+# Training workers can tolerate spot nodes
+apiVersion: v1
+kind: Pod
+metadata:
+  name: training-worker
+spec:
+  tolerations:
+  - key: node.kubernetes.io/sla
+    operator: Gt
+    value: "800"  # 80% SLA acceptable
+    effect: NoSchedule
+  containers:
+  - name: worker
+    resources:
+      requests:
+        nvidia.com/gpu: 4
+```
+
+#### Story 4 — DRA GPU claim management
+
+As a DRA driver implementer, I want to combine device resource claims with node SLA constraints so that GPU claims can only bind to nodes meeting a minimum reliability, unless the workload explicitly tolerates lower values.
+
+This ensures DRA allocations are both resource-correct and reliability-compliant.
+
+**Example Configuration:**
```

Comment:
Let's add the ResourceSlice example that includes the taint that ResourceClaim "gpu-claim-high-sla" will tolerate.

Reply:
Addressed